# Analysing Thanksgiving Dinner

Data Analysis on survey taken about Thanksgiving night

Source: FiveThirtyEight

# Importing and reading data

In [36]:
import pandas as pd
df=pd.read_csv('https://raw.githubusercontent.com/fivethirtyeight/data/master/thanksgiving-2015/thanksgiving-2015-poll-data.csv',encoding='latin-1')
df.head()

RespondentID Do you celebrate Thanksgiving?  \
0    4337954960                            Yes   
1    4337951949                            Yes   
2    4337935621                            Yes   
3    4337933040                            Yes   
4    4337931983                            Yes   

  What is typically the main dish at your Thanksgiving dinner?  \
0                                             Turkey             
1                                             Turkey             
2                                             Turkey             
3                                             Turkey             
4                                           Tofurkey             

  What is typically the main dish at your Thanksgiving dinner? - Other (please specify)  \
0                                                NaN                                      
1                                                NaN                                      
2                                                NaN                                      
3                                                NaN                                      
4                                                NaN                                      

  How is the main dish typically cooked?  \
0                                  Baked   
1                                  Baked   
2                                Roasted   
3                                  Baked   
4                                  Baked   

  How is the main dish typically cooked? - Other (please specify)  \
0                                                NaN                
1                                                NaN                
2                                                NaN                
3                                                NaN                
4                                                NaN                

  What kind of stuffing/dressing do you typically have?  \
0                                        Bread-based      
1                                        Bread-based      
2                                         Rice-based      
3                                        Bread-based      
4                                        Bread-based      

  What kind of stuffing/dressing do you typically have? - Other (please specify)  \
0                                                NaN                               
1                                                NaN                               
2                                                NaN                               
3                                                NaN                               
4                                                NaN                               

  What type of cranberry saucedo you typically have?  \
0                                               None   
1                             Other (please specify)   
2                                           Homemade   
3                                           Homemade   
4                                             Canned   

  What type of cranberry saucedo you typically have? - Other (please specify)  \
0                                                NaN                            
1                    Homemade cranberry gelatin ring                            
2                                                NaN                            
3                                                NaN                            
4                                                NaN                            

          ...          \
0         ...           
1         ...           
2         ...           
3         ...           
4         ...           

  Have you ever tried to meet up with hometown friends on Thanksgiving night?  \
0                                                Yes                            
1                                                 No                            


# Taking the rows only in which Thanksgiving celebrated is given 'Yes'

In [145]:
df['Do you celebrate Thanksgiving?'].value_counts()

Yes    980
Name: Do you celebrate Thanksgiving?, dtype: int64

In [147]:
df=df[df['Do you celebrate Thanksgiving?']=='Yes']

In [148]:
df.columns

Index(['RespondentID', 'Do you celebrate Thanksgiving?',
       'What is typically the main dish at your Thanksgiving dinner?',
       'What is typically the main dish at your Thanksgiving dinner? - Other (please specify)',
       'How is the main dish typically cooked?',
       'How is the main dish typically cooked? - Other (please specify)',
       'What kind of stuffing/dressing do you typically have?',
       'What kind of stuffing/dressing do you typically have? - Other (please specify)',
       'What type of cranberry saucedo you typically have?',
       'What type of cranberry saucedo you typically have? - Other (please specify)',
       'Do you typically have gravy?',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Brussel sprouts',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots',
       'Which of these side dishes aretypically served

# Main dishes at Thanksgiving dinner

In [41]:
df["What is typically the main dish at your Thanksgiving dinner?"].value_counts()

Turkey                    859
Other (please specify)     35
Ham/Pork                   29
Tofurkey                   20
Chicken                    12
Roast beef                 11
I don't know                5
Turducken                   3
Name: What is typically the main dish at your Thanksgiving dinner?, dtype: int64

It seems Turkey is the most common main dish at Thanksgiving dinner.

# Which dishes typically have gravy?

In [42]:
df[df['Do you typically have gravy?']=='Yes']["What is typically the main dish at your Thanksgiving dinner?"].value_counts()

Turkey                    814
Ham/Pork                   24
Other (please specify)     21
Tofurkey                   12
Chicken                    10
Roast beef                  7
I don't know                2
Turducken                   2
Name: What is typically the main dish at your Thanksgiving dinner?, dtype: int64

Almost all dishes have gravy with some exceptions in each dish.

# Finding how many people eat one of the Apple,Pecan and Pumpkin pies

In [43]:
apple_isnull=pd.isnull(df['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple'])
pecan_isnull=pd.isnull(df['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan'])
pumpkin_isnull=pd.isnull(df['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin'])
ate_pies=apple_isnull&pumpkin_isnull&pecan_isnull
ate_pies.value_counts()

False    876
True     104
dtype: int64

So 876 people eat at least one of the three types of pies

# Analyzing the Age column

As age column is categorical data type, will convert to numeric to analyze the ages of survey respondents

In [46]:
def refine_age(age):
    if pd.isnull(age):
        return None
    age=age.split(' ')[0]
    age=age.replace('+', '')
    return int(age)
df['Age']=df['Age'].apply(refine_age)
df['Age'].describe()

count    947.000000
mean      40.089757
std       15.352014
min       18.000000
25%       30.000000
50%       45.000000
75%       60.000000
max       60.000000
Name: Age, dtype: float64

# Findings

Although it is only a rough approximation of age, and as we took the first value in each string (the lower bound), it skews downward.

The age groups of respondents are also fairly distributed.

# Exploring Income of Respondents

In [50]:
df["How much total combined money did all members of your HOUSEHOLD earn last year?"].value_counts()

$25,000 to $49,999      166
$50,000 to $74,999      127
$75,000 to $99,999      127
Prefer not to answer    118
$100,000 to $124,999    109
$200,000 and up          76
$10,000 to $24,999       60
$0 to $9,999             52
$125,000 to $149,999     48
$150,000 to $174,999     38
$175,000 to $199,999     26
Name: How much total combined money did all members of your HOUSEHOLD earn last year?, dtype: int64

In [52]:
def refine_income(income_str):
    if pd.isnull(income_str):
        return None
    income_str = income_str.split(" ")[0]
    if income_str == "Prefer":
        return None
    income_str = income_str.replace(",", "")
    income_str = income_str.replace("$", "")
    return int(income_str)

df["int_income"] = df["How much total combined money did all members of your HOUSEHOLD earn last year?"].apply(refine_income)
df["int_income"].describe()

count       829.000000
mean      75965.018094
std       59068.636748
min           0.000000
25%       25000.000000
50%       75000.000000
75%      100000.000000
max      200000.000000
Name: int_income, dtype: float64

# Findings

Again it is only a rough approximation of income, and it skews downward as we took the first value in each string (the lower bound), the average income seems to be fairly high, although there is also a large standard deviation.

# Finding any relationship between Travel distance and Income

In [53]:
df[df["int_income"] < 150000]["How far will you travel for Thanksgiving?"].value_counts()

Thanksgiving is happening at my home--I won't travel at all                         281
Thanksgiving is local--it will take place in the town I live in                     203
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    150
Thanksgiving is out of town and far away--I have to drive several hours or fly       55
Name: How far will you travel for Thanksgiving?, dtype: int64

In [55]:
df[df["int_income"] > 150000]["How far will you travel for Thanksgiving?"].value_counts()

Thanksgiving is happening at my home--I won't travel at all                         49
Thanksgiving is local--it will take place in the town I live in                     25
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    16
Thanksgiving is out of town and far away--I have to drive several hours or fly      12
Name: How far will you travel for Thanksgiving?, dtype: int64

# Findings

People with low incomes mostly have thanksgiving at home or in local places than people with high incomes.

# Relating Friendship and Age

In [56]:
df.pivot_table(
    index="Have you ever tried to meet up with hometown friends on Thanksgiving night?", 
    columns='Have you ever attended a "Friendsgiving?"',
    values="Age"
)

Have you ever attended a "Friendsgiving?"                  No        Yes
Have you ever tried to meet up with hometown fr...                      
No                                                  42.283702  37.010526
Yes                                                 41.475410  33.976744

In [58]:
df.pivot_table(
    index="Have you ever tried to meet up with hometown friends on Thanksgiving night?", 
    columns='Have you ever attended a "Friendsgiving?"',
    values="int_income"
)

Have you ever attended a "Friendsgiving?"                     No           Yes
Have you ever tried to meet up with hometown fr...                            
No                                                  78914.549654  72894.736842
Yes                                                 78750.000000  66019.736842

# Findings

It appears that people who are younger are more likely to attend a Friendsgiving, and try to meet up with friends on Thanksgiving.

# Most common dessert people eat

In [75]:
df.iloc[:,39:48].notnull().sum().sort_values(ascending=False)

Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Ice cream        266
Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cookies          204
Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cheesecake       191
Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Brownies         128
Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Apple cobbler    110
Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Peach cobbler    103
Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Carrot cake       72
Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Fudge             43


So the most common desserts people eat are Ice cream,Cookies,Cheesecake

# Do people work on Thanksgiving

We have two survey questions related to this:
Do you work in retail?
Will you employer make you work on Black Friday?

Creating a pivot table makes it easy to know the count of people working in retail and on Black Friday

In [87]:
df.pivot_table(columns='Do you work in retail?',index='Will you employer make you work on Black Friday?',aggfunc='count',values='Age')

Do you work in retail?                            Yes
Will you employer make you work on Black Friday?     
Doesn't apply                                       7
No                                                 20
Yes                                                42

In the people who work in retail(70), 42 people work on Black Fridays!

# Finding regional patterns in the dinner menus.

In [104]:
df['US Region'].value_counts()   #Unique Regions in the data

South Atlantic        203
Middle Atlantic       145
East North Central    145
Pacific               130
West South Central     85
West North Central     71
East South Central     56
New England            55
Mountain               41
Name: US Region, dtype: int64

In [141]:
df_by_region=df.groupby('US Region').count()

# Count of people having the specific desserts grouped by region

In [142]:
df_by_region.iloc[:,39:48]

Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Apple cobbler  \
US Region                                                                                                                                   
East North Central                                                 13                                                                       
East South Central                                                  9                                                                       
Middle Atlantic                                                    23                                                                       
Mountain                                                            1                                                                       
New England                                                         2                                                                       
Pacific                                                            18                                                                       
South Atlantic                                                     24                                                                       
West North Central                                                  7                                                                       
West South Central                                                 11                                                                       

                    Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Blondies  \
US Region                                                                                                                              
East North Central                                                  1                                                                  
East South Central                                                  2                                                                  
Middle Atlantic                                                     0                                                                  
Mountain                                                            0                                                                  
New England                                                         6                                                                  
Pacific                                                             3                                                                  
South Atlantic                                                      3                                                                  
West North Central                                                  0                                                                  
West South Central                                                  1                                                                  

                    Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Brownies  \
US Region                                                                                                                              
East North Central                                                 16                                                                  
East South Central                                                  6                                                                  
Middle Atlantic                                                    18                                                                  
Mountain                                                            4                                                                  
New England                                                        11                                                                  
Pacific            

# Side dishes by region

In [143]:
df_by_region.iloc[:,11:24]

Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Brussel sprouts  \
US Region                                                                                                                                         
East North Central                                                 22                                                                             
East South Central                                                  7                                                                             
Middle Atlantic                                                    41                                                                             
Mountain                                                            6                                                                             
New England                                                        12                                                                             
Pacific                                                            29                                                                             
South Atlantic                                                     27                                                                             
West North Central                                                  3                                                                             
West South Central                                                  4                                                                             

                    Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots  \
US Region                                                                                                                                 
East North Central                                                 28                                                                     
East South Central                                                 15                                                                     
Middle Atlantic                                                    45                                                                     
Mountain                                                           11                                                                     
New England                                                        25                                                                     
Pacific                                                            30                                                                     
South Atlantic                                                     48                                                                     
West North Central                                                 12                                                                     
West South Central                                                 21                                                                     

                    Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Cauliflower  \
US Region                                                                                                                                     
East North Central                                                 13                                                                         
East South Central                                                  5                                                                         
Middle Atlantic                                                    25                                                                         
Mountain                                                            4                                                                         
New England   

# Pies by region

In [140]:
df_by_region.iloc[:,26:36]

Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple  \
US Region                                                                                                                      
East North Central                                                 75                                                          
East South Central                                                 19                                                          
Middle Atlantic                                                   106                                                          
Mountain                                                           19                                                          
New England                                                        41                                                          
Pacific                                                            74                                                          
South Atlantic                                                    101                                                          
West North Central                                                 37                                                          
West South Central                                                 32                                                          

                    Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Buttermilk  \
US Region                                                                                                                           
East North Central                                                  3                                                               
East South Central                                                  6                                                               
Middle Atlantic                                                     2                                                               
Mountain                                                            0                                                               
New England                                                         1                                                               
Pacific                                                             2                                                               
South Atlantic                                                      4                                                               
West North Central                                                  2                                                               
West South Central                                                 13                                                               

                    Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Cherry  \
US Region                                                                                                                       
East North Central                                                 22                                                           
East South Central                                                  4                                                           
Middle Atlantic                                                    17                                                           
Mountain                                                            7                                                           
New England                                                         0                                                           
Pacific                                                            20                                                           
South Atlantic                                                     20                                              

# Pivot table showing the Main dish and its count in each region

In [144]:
df.pivot_table(index='US Region',columns='What is typically the main dish at your Thanksgiving dinner?',aggfunc='count',values='Age')

What is typically the main dish at your Thanksgiving dinner?  Chicken  \
US Region                                                               
East North Central                                                NaN   
East South Central                                                NaN   
Middle Atlantic                                                   1.0   
Mountain                                                          1.0   
New England                                                       2.0   
Pacific                                                           NaN   
South Atlantic                                                    3.0   
West North Central                                                1.0   
West South Central                                                1.0   

What is typically the main dish at your Thanksgiving dinner?  Ham/Pork  \
US Region                                                                
East North Central                                                 4.0   
East South Central                                                 1.0   
Middle Atlantic                                                    2.0   
Mountain                                                           1.0   
New England                                                        NaN   
Pacific                                                            6.0   
South Atlantic                                                     7.0   
West North Central                                                 4.0   
West South Central                                                 2.0   

What is typically the main dish at your Thanksgiving dinner?  I don't know  \
US Region                                                                    
East North Central                                                     NaN   
East South Central                                                     NaN   
Middle Atlantic                                                        NaN   
Mountain                                                               NaN   
New England                                                            NaN   
Pacific                                                                1.0   
South Atlantic                                                         NaN   
West North Central                                                     1.0   
West South Central                                                     NaN   

What is typically the main dish at your Thanksgiving dinner?  Other (please specify)  \
US Region                                                                              
East North Central                                                               5.0   
East South Central                                                               4.0   
Middle Atlantic                                                                  4.0   
Mountain                                                                         NaN   
New England                                                                      1.0   
Pacific                                                                          9.0   
South Atlantic                                                                   6.0   
West North Central                                                               3.0   
West South Central                                                               3.0   

What is typically the main dish at your Thanksgiving dinner?  Roast beef  \
US Region                                                                  
East North Central                                                   NaN   
East South Central                                                   1.0   
Middle Atlantic                                                      2.0   
Mountain                                                             NaN   
New England                                                          NaN   
Pacific              

# Findings

Turkey is typically the main dish in all regions with highest in South Atlantic and lowest in East South Central and New England.